lib/Lift/InstructionTranslator.cpp

```diff
@@ -1202,48 +1134,57 @@ IT::translateOpcode(PTCOpcode Opcode,
     Value *Truncated = Builder.CreateTrunc(InArguments[0], SourceType);
 
     switch (Opcode) {
-    case PTC_INSTRUCTION_op_ext8s_i32:
-    case PTC_INSTRUCTION_op_ext8s_i64:
-    case PTC_INSTRUCTION_op_ext16s_i32:
-    case PTC_INSTRUCTION_op_ext16s_i64:
-    case PTC_INSTRUCTION_op_ext32s_i64:
+    case LIBTCG_op_ext8s_i32:
+    case LIBTCG_op_ext8s_i64:
+    case LIBTCG_op_ext16s_i32:
+    case LIBTCG_op_ext16s_i64:
+    case LIBTCG_op_ext32s_i64:
+    case LIBTCG_op_ext_i32_i64:
       return v{ Builder.CreateSExt(Truncated, RegisterType) };
-    case PTC_INSTRUCTION_op_ext8u_i32:
-    case PTC_INSTRUCTION_op_ext8u_i64:
-    case PTC_INSTRUCTION_op_ext16u_i32:
-    case PTC_INSTRUCTION_op_ext16u_i64:
-    case PTC_INSTRUCTION_op_ext32u_i64:
+    case LIBTCG_op_ext8u_i32:
+    case LIBTCG_op_ext8u_i64:
+    case LIBTCG_op_ext16u_i32:
+    case LIBTCG_op_ext16u_i64:
+    case LIBTCG_op_ext32u_i64:
+    case LIBTCG_op_extu_i32_i64:
       return v{ Builder.CreateZExt(Truncated, RegisterType) };
     default:
       revng_unreachable("Unexpected opcode");
     }
   }
-  case PTC_INSTRUCTION_op_not_i32:
-  case PTC_INSTRUCTION_op_not_i64:
+  case LIBTCG_op_extrl_i64_i32: {
+      return v{ Builder.CreateTrunc(InArguments[0], Builder.getInt32Ty()) };
+  }
+  case LIBTCG_op_extrh_i64_i32: {
+      Value *Shifted = Builder.CreateAShr(InArguments[0], ConstantInt::get(Builder.getInt64Ty(), 32));
+      return v{ Builder.CreateTrunc(Shifted, Builder.getInt32Ty()) };
+  }
```

Comment:
These are new instructions right?
Can you make a list of new instructions so I can stare at them intensely for a while?

Reply:
The major commit which updates InstructionTranslator is only to maintain parity in terms of instruction support. All new instructions are handled in separate following commits. Look at:

  "Lift/InstructionTranslator: handle ext[rl] + exitTB",
  "Lift/InstructionTranslator: handle clz_i[32|64],[s]extract[2]_i32",
  "Lift/InstructionTranslator: handle [s]extract_i64"
  "Lift/InstructionTranslator: handle ctz_i[32|64],negsetcond_[32|64]"